app/uk/gov/hmrc/agentsubscription/controllers/SubscriptionController.scala

```diff
@@ -44,4 +44,17 @@ class SubscriptionController @Inject() (subscriptionService: SubscriptionService
       }
     }
   }
+
+  def updateSubscription = affinityGroupAndEnrolments { implicit request => implicit authIds =>
+    implicit val hc = fromHeadersAndSession(request.headers, None)
```

Comment:
fromHeadersAndSession name does not mean much, what does it do rename to getHeaderCarriercFromHeadersAndSession